subsys/bluetooth/host/smp.c

```diff
@@ -798,7 +798,13 @@ static void smp_br_timeout(struct k_work *work)
 static void smp_br_send(struct bt_smp_br *smp, struct net_buf *buf,
 			bt_conn_tx_cb_t cb)
 {
-	if (bt_l2cap_send_cb(smp->chan.chan.conn, BT_L2CAP_CID_BR_SMP, buf, cb, NULL)) {
+	int err = bt_l2cap_send_cb(smp->chan.chan.conn, BT_L2CAP_CID_BR_SMP, buf, cb, NULL);
+
+	if (err) {
+		if (err == -ENOBUFS) {
+			LOG_ERR("Ran out of TX buffers or contexts.");
+		}
```

Comment:
Since this pattern appear 3 times now, perhaps the LOG_ERR message should be put into `bt_l2cap_send_cb` instead?

i.e. 
```c
int bt_l2cap_send_cb(struct bt_conn *conn, uint16_t cid, struct net_buf *buf,
		     bt_conn_tx_cb_t cb, void *user_data)
{
	struct bt_l2cap_hdr *hdr;
	int err;

	LOG_DBG("conn %p cid %u len %zu", conn, cid, net_buf_frags_len(buf));

	hdr = net_buf_push(buf, sizeof(*hdr));
	hdr->len = sys_cpu_to_le16(buf->len - sizeof(*hdr));
	hdr->cid = sys_cpu_to_le16(cid);

	err = bt_conn_send_cb(conn, buf, cb, user_data);
	if (err == -ENOBUFS) {
		LOG_ERR("Ran out of TX buffers or contexts.");
	}
	
	return err;
}
```

Reply:
> no that's the point. L2CAP shouldn't care about this error, it's the upper layers' job to do something about it.

This is a matter of where a LOG message is being done; it doesn't affect anything. Putting the error message in ATT, instead of L2CAP, does not help the upper layers to do anything (that's the purpose of the return code).  

But I see what you mean. 

One question: If a developer (who can actually do anything about it) see's this error, can the developer solve the issue without changing the stack? I assume they can increase the number of buffers or something, right? 

When/if `bt_l2cap_send_cb` fails, is that return error propagated to the upper layers? If so, then this should not have been a LOG_ERR in any case, and should be a LOG_DBG IMO.

Reply:
yes and yes. but some layers just ignore it (cf SMP). That's why I added a log err there.